main.go

```diff
@@ -125,7 +135,10 @@ var rootCmd = &coral.Command{
 			name:     name,
 			start:    time.Now(),
 		}
-		return tea.NewProgram(m).Start()
+        // the default handler exits with 0
+        // without the default handler it exits with 1
+        p := tea.NewProgram(m, tea.WithoutSignalHandler())
```

Comment:
without signal handler afaik only disables the SIGWICH signal, not SIGINT/SIGTERM, so its not really needed.

That said, probably `os.Exit(1)` is not the best thing to do in the `Update` method, I'll take a look.

Reply:
Thanks for the feedback appreciate it!

```go
// Listen for SIGINT and SIGTERM.
//
// In most cases ^C will not send an interrupt because the terminal will be
// in raw mode and ^C will be captured as a keystroke and sent along to
// Program.Update as a KeyMsg. When input is not a TTY, however, ^C will be
// caught here.
//
// SIGTERM is sent by unix utilities (like kill) to terminate a process.
```
I oriented on that comment in the [tea.go](https://github.com/charmbracelet/bubbletea/blob/master/tea.go) and the handler sends a quit-message which then leads to a exit(0).
Without the handler the SIGTERM/ SIGINT sent from e.g. `killall timer` ends the program with exit(1).

Well if understood that right I am not sure...

Reply:
I am gone dig a little bit more into that topic looking forward to what you come up with :) :+1:

Reply:
I did basically the same thing, except that I don't exit 1, I set an interrupt bool in the model and return an error later, which in turn exits 1.

its just that exiting 1 without the tea app cleaning up after itself might leave the term in a "broken" state.

Reply:
Ohh ok, **thank you very much for the explanation** :+1: . Yeah that makes sense.